java/datastore/src/test/java/com/google/datastore/v1/client/RemoteRpcTest.java

```diff
@@ -132,6 +139,25 @@ public void testGzip() throws IOException, DatastoreException {
     assertEquals(-1, injectedTestValues.inputStream.read());
   }
 
+  @Test
+  public void testHttpHeaders() throws IOException {
+    RollbackRequest.Builder rollBackReq = RollbackRequest.newBuilder();
+    rollBackReq.setTransaction(ByteString.copyFromUtf8("project-id"));
+    MessageLite request = rollBackReq.build();
+    ProtoHttpContent payload = new ProtoHttpContent(request);
+    BeginTransactionResponse response = newBeginTransactionResponse();
+    InjectedTestValues injectedTestValues =
+        new InjectedTestValues(gzip(response), new byte[1], true);
+    RemoteRpc rpc = newRemoteRpc(injectedTestValues);
+    HttpRequest httpRequest = rpc.getClient().buildPostRequest(rpc.resolveURL("blah"),
+        payload);
+    rpc.setHeaders(request, httpRequest);
+    String header = httpRequest.getHeaders().getFirstHeaderStringValue(
+        EndToEndChecksumHandler.HTTP_REQUEST_CHECKSUM_HEADER);
+    assertNotNull(header);
```

Comment:
Can we make this assertion stronger?

Reply:
I have added another check below this: assertFalse(header.isEmpty())
is that enough?

If we can bump the JDK target level in pom.xml to 11, then I can use 
    assertFalse(header.isBlank())

or any other suggestions please?
( I am trying to avoid verification of the exact checksum value - unless you think we should do it)

Reply:
Here and in previous assertions, if you are later asserting properties on the object, we can remove the `assertNotNull` as it is now implied

Reply:
Done